011-php-code-formatting.md

```diff
@@ -0,0 +1,94 @@
+
+The way the plugin is configured, in daily use developers will not need to manually run the command to format their code since it will run as a `pre-commit` hook on any changed files.
+
+Additionally, some code editors (like Visual Studio Code) easily support running Prettier on file saves (refer to the code editor docs for documentation on setup).
+
+### PHP Code Standards Fixer
+
+The [PHP Coding Standards Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) tool cares less about the "look" of the PHP Code, and more about the code quality. It focuses on automatically fixing PHP code to follow a specific set of code quality rules, and in some scenarios it can even modernize and optimize the code.
+
+It helps developers eliminate any concerns regarding the quality of their PHP code, whether they are adhering to [PSR](https://www.php-fig.org/psr/) standards and helps with linting their code.
+
+You can see a [pull request](https://github.com/DoSomething/rogue/pull/1092) on our Rogue application with how the PHP CS Fixer tool is configured and how it proceeded to format the code.
+
+This [commit](https://github.com/DoSomething/rogue/commit/eeca6efb6b29df01e13e816b7d051e872b8cd2f7) specifically shows how the PHP CS Fixer tool is set up. The process is simplified by utilizing the [weerd/php-style](https://github.com/weerd/php-style) package which contains easily accessible rulesets for both generic PHP packages and Laravel projects. This package is also version controlled, so developers can rest easy knowing they can specify a version of the ruleset to use and they can opt-in to ruleset changes by adjusting their composer package version. This scenario has been a pain point for the team in the past that affected us on [more than one](https://dosomething.slack.com/archives/C3ASB4204/p1572615964090600) [occasion](https://dosomething.slack.com/archives/CUQMU4Q6B/p1596116997002700?thread_ts=1596116997.002700), where StyleCI made a change and affected all our repositories since their rules are not version controlled.
+
+This [commit](https://github.com/DoSomething/rogue/pull/1092/commits/d48ec9a1a759941c49c5e687a80b639e9d808fed) specifically shows all the files changed when the `composer format` command was executed within the Homestead vagrant environment.
+
+Since this tool is run as a composer command, based on most of our development setups, the command should be run within the Homestead vagrant box. In practice, developers can potentially run the command manually, but it is also setup to run within our Continuous Integration environment and it will flag any failures within a pull request (similar to how StyleCI behaves).
+
+However, once flagged in a pull request, to automatically fix these errors, developers can just run the `composer format` command in Homestead and then commit and push up their code, confident that the errors were addressed and the style checks should pass.
```

Comment:
Is the main difference between this and StyleCI in practice then that when I push up a PR that gets an ❌ , I can run a command to automatically fix the issues?

Reply:
Ya, you'll see the failure in the PR with an ❌, but instead of having to jump into StyleCI and interpret what it's telling you to change and hopefully understanding, with PHP CS Fixer, you just run `composer format` in your Homestead, push up the changes and PR should pass, easy-peasy 🍋 🗜

Reply:
GREAT emojis!!! Okay I dig that! Especially since it won't be for _everything_ because hopefully Prettier can handle most of it!